src/sysl/importers/import_swagger.py

```diff
@@ -94,200 +76,287 @@ def javaParam(match):
+        ident = self._param_cache.get(param)
+
+        if ident is None:
+            # foo-bar to fooBar
+            ident = re.sub(r'-(\w?)', lambda m: m.group(1).upper(), param)
+
+            # {fooid} -> fooId (only if foo is in WORDS but fooid isn't)
+            m = re.match(r'{([a-z]+)id}$', ident)
+            if m:
+                word = m.group(1)
+                if word in self.words() and word + 'id' not in self.words():
+                    ident = '{' + word + 'Id}'
+
+            ident = ident.replace('}', '<:string}')
+
+            self._param_cache[param] = ident
+
+        return ident
+
+    def translate_path_template_params(self, path):
+        # OAS 2.0 supports path templating.
+        # ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#pathTemplating
+        # For some reason, we rename these params via javaParam
+        return re.sub(r'({[^/]*?})', self.javaParam, path)
+
+    def translate(self, swag, appname, package, w):
+
+        if 'info' in swag:
+            def info_attrs(info, prefix=''):
+                for (name, value) in sorted(info.iteritems()):
+                    if isinstance(value, dict):
+                        info_attrs(value, prefix + name + '.')
+                    else:
+                        w('@{}{} = {}', prefix, name, json.dumps(value))
+
+            title = swag['info'].pop('title', '')
+            info_attrs(swag['info'])
+        else:
+            title = ''
+
+        if 'host' in swag:
+            w('@host = {}', json.dumps(swag['host']))
+
+        w(u'{}{} [package={}]:',
+            appname, title and ' ' + json.dumps(title), json.dumps(package))
+
+        with w.indent():
+            w(u'| {}', swag['info'].get('description', 'No description.'))
+
+            for (path, api) in sorted(swag['paths'].iteritems()):
+                w(u'\n{}:', self.translate_path_template_params(path))
+                with w.indent():
+                    if 'parameters' in api:
+                        del api['parameters']
+                    for (i, (method, body)) in enumerate(sorted(api.iteritems(),
+                                                                key=lambda t: METHOD_ORDER[t[0]])):
+                        qparams = dict()
+
+                        if 'parameters' in body:
+                            qparams = [p for p in body['parameters']
+                                       if p.get('in') == 'query']
+
+                        w(u'{}{}{}:',
+                          method.upper(),
+                          ' ?' if qparams else '',
+                          '&'.join(('{}={}{}'.format(p['name'],
+                                                     SWAGGER_TYPE_MAP[p['type']],
+                                                     '' if p['required'] else '?')
+                                    if p['type'] != 'string' else '{name}=string'.format(**p))
+                                   for p in qparams))
```

Comment:
This is ugly, but it's not obvious how to make it prettier. Here's my take. Take it or leave it.
```python
                        w(u'{}{}{}:',
                          method.upper(),
                          ' ?' if qparams else '',
                          '&'.join(
                            '{name}=string'.format(**p)
                            if p['type'] == 'string'
                            else '{}={}{}'.format(
                              p['name'],
                              SWAGGER_TYPE_MAP[p['type']],
                              '' if p['required'] else '?',
                            )
                            for p in qparams
                          )
                        )
```